### PR TITLE
Use new UI for `this.`/`Me.` and add a fix-all code action for qualification.

### PR DESCRIPTION
**Pull request by brettfo**
### High Level

This PR updates the `this.`/`Me.` code style options to support the requested notification levels (`None`, `Info`, `Warning`, `Error`) and also enables a fix-all code action.

### Details

- The existing code style options needed to be moved down from the features layer to the workspaces layer because that's where the analyzer and fixer for `this.`/`Me.` live.
- `SimpleCodeStyleOption` was renamed to `CodeStyleOption<T>` to allow for any option value type, although only `bool` is currently used.
- The options UI now uses the new option and notification level and all of that is serialized appropriately and honored by the analyzer.

### Contentious points

- The old options (e.g., `SimplificationOptions.QualifyFieldAccess`) were obsoleted in favor of the new options (e.g., `CodeStyleOptions.QualifyFieldAccess`) because the old options were only simple `bool`s, but the new ones also manage their notification level.  **With this obsoletion, do we need to provide a way for the old option to migrate forward?  Or since we're targeting a future release of VS can we simply delete the old options from the API?**  If we have to support the old options still, what should happen to the notification preference if I set the old option value to `false`.  Does that mean `NotificationOptions.None` or `Error` or something more complicated?
- There is also the possibility of leaving the existing `SimplificationOptions.QualifyFieldAccess`, etc. and supplementing them with additional options like `CodeStyleOptions.QualifyFieldAccessNotification`, but that feels weird for two reasons, **1)** the options are coupled in the UI but de-coupled in the API; what's the mechanism for discoverability between the two?  And **2)** the existing `var`/keyword, etc., options are already coupled together and the inconsistency might kill me.

### Still pending

- Update internal integration tests.

### CODE REVIEW

**Comment by brettfo** [comment]
@dotnet/roslyn-ide for review

File: src/Features/Core/Portable/Diagnostics/Analyzers/SimplifyTypeNamesDiagnosticAnalyzerBase.cs [hunk 1]
```diff
@@ -33,19 +34,46 @@ internal abstract class SimplifyTypeNamesDiagnosticAnalyzerBase<TLanguageKindEnu
                                                                     customTags: DiagnosticCustomTags.Unnecessary);
 
         private static readonly LocalizableString s_localizableTitleRemoveThisOrMe = new LocalizableResourceString(nameof(FeaturesResources.RemoveQualification), FeaturesResources.ResourceManager, typeof(FeaturesResources));
-        private static readonly DiagnosticDescriptor s_descriptorRemoveThisOrMe = new DiagnosticDescriptor(IDEDiagnosticIds.RemoveQualificationDiagnosticId,
+        private static readonly DiagnosticDescriptor s_descriptorRemoveThisOrMeHidden = new DiagnosticDescriptor(IDEDiagnosticIds.RemoveQualificationDiagnosticId,
```
Comment (Pilchie) [inline src/Features/Core/Portable/Diagnostics/Analyzers/SimplifyTypeNamesDiagnosticAnalyzerBase.cs]: If you look at the naming styles stuff, I think you can have a single `DiagnosticDescriptor` and then return a different severity later  (Just be careful, if your "defaultSeverity" is Hidden, then the anlayzer won't be run on closed files).

**Comment by Pilchie** [comment]
:+1:  Anyone else from @dotnet/roslyn-ide?

**Comment by brettfo** [comment]
Rebased to fix merge conflicts.

If nobody else has any complaints imma merge.

<sub>Not really; I'm just trying to scare some reviews out of people.</sub>

**Comment by brettfo** [comment]
test vsi please

**Comment by brettfo** [comment]
FYI, the internal component of this PR is dotnet/roslyn-internal#1000.

**Pull request merged**